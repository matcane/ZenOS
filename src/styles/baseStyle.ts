import { StyleSheet } from "react-native";

export const baseStyle = StyleSheet.create({
  transparent: { backgroundColor: undefined },

  fontXS: { fontSize: 12 },
  fontSM: { fontSize: 16 },
  fontSMD: { fontSize: 24 },
  fontMD: { fontSize: 32 },
  fontLG: { fontSize: 48 },
  fontXL: { fontSize: 64 },
  fontXXL: { fontSize: 128 },

  heightSM: { height: 16 },
  heightMD: { height: 32 },
  heightLG: { height: 48 },
  heightXL: { height: 64 },
  heightXXL: { height: 128 },

  widthSM: { width: 16 },
  widthMD: { width: 32 },
  widthLG: { width: 48 },
  widthXL: { width: 64 },
  widthXXL: { width: 128 },

  sizeSM: { width: 16, height: 16 },
  sizeMD: { width: 32, height: 32 },
  sizeLG: { width: 48, height: 48 },
  sizeXL: { width: 64, height: 64 },

  static: { position: "static" },
  absolute: { position: "absolute" },
  relative: { position: "relative" },

  rounded: { borderRadius: 5 },
  roundedMD: { borderRadius: 10 },
  roundedLG: { borderRadius: 20 },
  roundedFull: { borderRadius: 999 },

  flex: { flex: 1 },
  flexNone: { flex: 0 },
  flexGrow: { flexGrow: 1 },
  flexGrowNone: { flexGrow: 0 },
  flexShrink: { flexShrink: 1 },
  flexShrinkNone: { flexShrink: 0 },

  flexRow: { flexDirection: "row" },
  flexRowReverse: { flexDirection: "row-reverse" },
  flexCol: { flexDirection: "column" },
  flexColReverse: { flexDirection: "column-reverse" },

  paddingSM: { padding: 5 },
  paddingMD: { padding: 10 },
  paddingLG: { padding: 15 },
  paddingXL: { padding: 20 },

  paddingHorizontalSM: { paddingHorizontal: 10 },
  paddingHorizontalMD: { paddingHorizontal: 20 },
  paddingHorizontalLG: { paddingHorizontal: 30 },
  paddingHorizontalXL: { paddingHorizontal: 40 },
  paddingHorizontalXXL: { paddingHorizontal: 90 },

  paddingVerticalSM: { paddingVertical: 10 },
  paddingVerticalMD: { paddingVertical: 20 },
  paddingVerticalLG: { paddingVertical: 30 },
  paddingVerticalXL: { paddingVertical: 40 },
  paddingVerticalXXL: { paddingVertical: 80 },

  paddingTopSM: { paddingTop: 5 },
  paddingTopMD: { paddingTop: 10 },
  paddingTopLG: { paddingTop: 15 },
  paddingTopXL: { paddingTop: 20 },
  paddingTopXXL: { paddingTop: 40 },

  paddingBottomSM: { paddingBottom: 5 },
  paddingBottomMD: { paddingBottom: 10 },
  paddingBottomLG: { paddingBottom: 15 },
  paddingBottomXL: { paddingBottom: 20 },
  paddingBottomXXL: { paddingBottom: 40 },

  paddingLeftSM: { paddingLeft: 5 },
  paddingLeftMD: { paddingLeft: 10 },
  paddingLeftLG: { paddingLeft: 15 },
  paddingLeftXL: { paddingLeft: 20 },
  paddingLeftXXL: { paddingLeft: 40 },

  paddingRightSM: { paddingRight: 5 },
  paddingRightMD: { paddingRight: 10 },
  paddingRightLG: { paddingRight: 15 },
  paddingRightXL: { paddingRight: 20 },
  paddingRightXXL: { paddingRight: 40 },

  marginSM: { margin: 5 },
  marginMD: { margin: 10 },
  marginLG: { margin: 15 },
  marginXL: { margin: 20 },

  marginHorizontalSM: { marginHorizontal: 10 },
  marginHorizontalMD: { marginHorizontal: 20 },
  marginHorizontalLG: { marginHorizontal: 30 },
  marginHorizontalXL: { marginHorizontal: 40 },
  marginHorizontalXXL: { marginHorizontal: 80 },

  marginVerticalSM: { marginVertical: 10 },
  marginVerticalMD: { marginVertical: 20 },
  marginVerticalLG: { marginVertical: 30 },
  marginVerticalXL: { marginVertical: 40 },
  marginVerticalXXL: { marginVertical: 80 },

  marginTopSM: { marginTop: 5 },
  marginTopMD: { marginTop: 10 },
  marginTopLG: { marginTop: 15 },
  marginTopXL: { marginTop: 20 },
  marginTopXXL: { marginTop: 40 },

  marginBottomSM: { marginBottom: 5 },
  marginBottomMD: { marginBottom: 10 },
  marginBottomLG: { marginBottom: 15 },
  marginBottomXL: { marginBottom: 20 },
  marginBottomXXL: { marginBottom: 40 },

  marginLeftSM: { marginLeft: 5 },
  marginLeftMD: { marginLeft: 10 },
  marginLeftLG: { marginLeft: 15 },
  marginLeftXL: { marginLeft: 20 },
  marginLeftXXL: { marginLeft: 40 },

  marginRightSM: { marginRight: 5 },
  marginRightMD: { marginRight: 10 },
  marginRightLG: { marginRight: 15 },
  marginRightXL: { marginRight: 20 },
  marginRightXXL: { marginRight: 40 },

  justifyStart: { justifyContent: "flex-start" },
  justifyEnd: { justifyContent: "flex-end" },
  justifyCenter: { justifyContent: "center" },
  justifyBetween: { justifyContent: "space-between" },
  justifyAround: { justifyContent: "space-around" },
  justifyEvenly: { justifyContent: "space-evenly" },

  contentStart: { alignContent: "flex-start" },
  contentEnd: { alignContent: "flex-end" },
  contentCenter: { alignContent: "center" },
  contentBetween: { alignContent: "space-between" },
  contentAround: { alignContent: "space-around" },
  contentEvenly: { alignContent: "space-evenly" },

  itemsStart: { alignItems: "flex-start" },
  itemsEnd: { alignItems: "flex-end" },
  itemsCenter: { alignItems: "center" },
  itemsBaseline: { alignItems: "baseline" },
  itemsStretch: { alignItems: "stretch" },

  textLeft: { textAlign: "left" },
  textCenter: { textAlign: "center" },
  textRight: { textAlign: "right" },
  textJustify: { textAlign: "justify" },
});
