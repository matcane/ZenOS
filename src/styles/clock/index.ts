import { StyleSheet } from "react-native";

export const clockStyles = StyleSheet.create({
  stopwatchStartStop: { right: "50%", marginRight: -30, bottom: 0, marginBottom: 30 },

  stopwatchReset: { right: "25%", marginRight: -20, bottom: 0, marginBottom: 35 },
});
