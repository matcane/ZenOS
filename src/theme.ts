export const Colors = {
  light: {
    text: "#212121",
    primary: "#2196f3",
    container: "#e0e0e0",
    secondaryContainer: "#a0a0a0",
    background: "#FFFFFF",
    invBackground: "#000000",
  },
  dark: {
    text: "#ffffff",
    primary: "#2196f3",
    container: "#212121",
    secondaryContainer: "#404040",
    background: "#000000",
    invBackground: "#FFFFFF",
  },
};
