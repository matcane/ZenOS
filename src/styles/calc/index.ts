import { StyleSheet } from "react-native";

import { baseStyle } from "../baseStyle";

export const calcStyles = StyleSheet.create({
  calcDisplay: {
    ...baseStyle.itemsEnd,
    ...baseStyle.justifyEnd,
    ...baseStyle.paddingBottomXL,
  },

  calcDisplayText: { ...baseStyle.paddingHorizontalMD },
});
