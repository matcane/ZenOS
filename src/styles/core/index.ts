import { StyleSheet } from "react-native";

import { baseStyle } from "../baseStyle";

export const coreStyles = StyleSheet.create({
  calcDisplay: {
    ...baseStyle.itemsEnd,
    ...baseStyle.justifyEnd,
    ...baseStyle.paddingBottomXL,
  },

  calcDisplayText: { ...baseStyle.paddingHorizontalMD },

  stopwatchStartStop: { right: "50%", marginRight: -30, bottom: 0, marginBottom: 30 },

  stopwatchReset: { right: "25%", marginRight: -20, bottom: 0, marginBottom: 35 },

  actionButton: {
    ...baseStyle.absolute,
    ...baseStyle.roundedFull,
    ...baseStyle.justifyCenter,
    ...baseStyle.itemsCenter,
    elevation: 5,
    shadowColor: "#000",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.8,
    shadowRadius: 2,
  },

  container: {
    ...baseStyle.paddingXL,
    ...baseStyle.itemsCenter,
  },

  navigationBarContainer: {
    ...baseStyle.justifyEvenly,
    ...baseStyle.itemsCenter,
    ...baseStyle.flexRow,
    height: 45,
    gap: 20,
  },

  backNavButton: { transform: [{ rotate: "30deg" }] },

  settingsFieldItem: {
    ...baseStyle.paddingLeftMD,
    ...baseStyle.flexRow,
    ...baseStyle.justifyBetween,
    ...baseStyle.itemsCenter,
    ...baseStyle.flexGrow,
    height: 60,
    gap: 10,
  },

  firstSettingsFieldItem: {
    borderTopEndRadius: 10,
    borderTopStartRadius: 10,
  },

  lastSettingsFieldItem: {
    borderBottomEndRadius: 10,
    borderBottomStartRadius: 10,
  },

  settingsFieldIcon: {
    ...baseStyle.sizeLG,
  },

  statusBarContainer: {
    ...baseStyle.absolute,
    ...baseStyle.flexRow,
    ...baseStyle.justifyBetween,
    ...baseStyle.itemsCenter,
    ...baseStyle.paddingHorizontalMD,
    height: 32,
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    zIndex: 999,
  },

  statusBarIcons: {
    ...baseStyle.flexRow,
    gap: 2,
  },

  wallpaper: {
    ...baseStyle.flexGrow,
    ...baseStyle.justifyCenter,
    ...baseStyle.paddingTopXL,
  },
});
