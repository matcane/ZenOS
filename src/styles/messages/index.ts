import { StyleSheet } from "react-native";

import { baseStyle } from "../baseStyle";

export const messagesStyles = StyleSheet.create({
  chatInputContainer: {
    ...baseStyle.absolute,
    ...baseStyle.flexRow,
    ...baseStyle.itemsCenter,
    columnGap: 20,
    bottom: 0,
  },
  chatInput: {
    ...baseStyle.paddingLeftLG,
    ...baseStyle.marginLeftLG,
    ...baseStyle.flexGrow,
    ...baseStyle.roundedLG,
  },
  chatInputButton: {
    ...baseStyle.paddingMD,
    ...baseStyle.marginVerticalSM,
    ...baseStyle.marginRightLG,
    ...baseStyle.roundedMD,
  },
  chatMessage: {
    maxWidth: "90%",
    ...baseStyle.textLeft,
    ...baseStyle.roundedLG,
    ...baseStyle.paddingLG,
  },
  chatMessageInfoButton: {
    ...baseStyle.justifyAround,
    ...baseStyle.marginLG,
    ...baseStyle.roundedLG,
    ...baseStyle.heightXL,
    ...baseStyle.flexRow,
  },
  chatMessageInfoAvatar: {
    ...baseStyle.transparent,
    ...baseStyle.paddingSM,
    ...baseStyle.marginLeftSM,
  },
  chatMessageInfoBody: {
    ...baseStyle.flexRow,
    ...baseStyle.justifyBetween,
    ...baseStyle.transparent,
    ...baseStyle.marginRightMD,
  },
  newMessage: { right: "5%", bottom: "2%" },
});
