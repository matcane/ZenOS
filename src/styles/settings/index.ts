import { StyleSheet } from "react-native";

import { baseStyle } from "../baseStyle";

export const settingsStyles = StyleSheet.create({
  settingsFieldItem: {
    ...baseStyle.paddingLeftMD,
    ...baseStyle.flexRow,
    ...baseStyle.justifyBetween,
    ...baseStyle.itemsCenter,
    ...baseStyle.flexGrow,
    height: 60,
    gap: 10,
  },

  firstSettingsFieldItem: {
    borderTopEndRadius: 10,
    borderTopStartRadius: 10,
  },

  lastSettingsFieldItem: {
    borderBottomEndRadius: 10,
    borderBottomStartRadius: 10,
  },

  settingsFieldIcon: {
    ...baseStyle.sizeLG,
  },
});
