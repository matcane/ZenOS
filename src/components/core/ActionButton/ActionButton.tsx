import MaterialCommunityIcons from "@expo/vector-icons/MaterialCommunityIcons";
import { StyleProp, TouchableOpacity, ViewStyle } from "react-native";

import { useTheme } from "@/hooks/core";
import { baseStyle } from "@/styles/baseStyle";
import { coreStyles } from "@/styles/core";

const { sizeMD, sizeLG, sizeXL } = baseStyle;
const { actionButton } = coreStyles;

type TIconName = "play" | "pause" | "repeat";
type TVariants = "primary" | "secondary";

type ActionButtonProps = {
  variant: TVariants;
  active: boolean;
  iconName: TIconName;
  style: StyleProp<ViewStyle>;
  fn: () => void;
};

export default function ActionButton({ variant, active, iconName, style, fn }: ActionButtonProps) {
  const theme = useTheme();
  const buttonSize = variant === "primary" ? sizeXL : sizeLG;
  const iconSize = variant === "primary" ? sizeLG : sizeMD;
  const backgroudColor = variant === "primary" ? theme.primary : theme.container;

  if (!active) return null;

  return (
    <TouchableOpacity
      testID="action-button"
      onPress={active && fn}
      style={[actionButton, buttonSize, { backgroundColor: backgroudColor }, style]}>
      <MaterialCommunityIcons name={iconName} size={iconSize.height} color="white" />
    </TouchableOpacity>
  );
}
