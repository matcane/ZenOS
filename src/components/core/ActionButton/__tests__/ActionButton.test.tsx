import MaterialCommunityIcons from "@expo/vector-icons/MaterialCommunityIcons";
import { render, fireEvent } from "@testing-library/react-native";

import { baseStyle } from "@/styles/baseStyle";
import { Colors } from "@/theme";

import ActionButton from "../ActionButton";

const { sizeMD, sizeLG, sizeXL } = baseStyle;

describe("ActionButton component", () => {
  it("renders correctly", () => {
    const fn = jest.fn();
    const tree = render(
      <ActionButton variant="primary" active={true} iconName="play" style={{}} fn={fn} />,
    );

    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with primary variant", () => {
    const fn = jest.fn();
    const { getByTestId } = render(
      <ActionButton variant="primary" active={true} iconName="play" style={{}} fn={fn} />,
    );
    const button = getByTestId("action-button");
    const buttonIcon = button.findByType(MaterialCommunityIcons);
    expect(button.props.style.backgroundColor).toBe(Colors.light.primary);
    expect(button.props.style.width).toBe(sizeXL.width);
    expect(button.props.style.height).toBe(sizeXL.height);
    expect(buttonIcon.props.size).toBe(sizeLG.height);
  });

  it("renders correctly with secondary variant", () => {
    const fn = jest.fn();
    const { getByTestId } = render(
      <ActionButton variant="secondary" active={true} iconName="pause" style={{}} fn={fn} />,
    );
    const button = getByTestId("action-button");
    const buttonIcon = button.findByType(MaterialCommunityIcons);
    expect(button.props.style.backgroundColor).toBe(Colors.light.container);
    expect(button.props.style.width).toBe(sizeLG.width);
    expect(button.props.style.height).toBe(sizeLG.height);
    expect(buttonIcon.props.size).toBe(sizeMD.height);
  });

  it("does not render when active is false", () => {
    const fn = jest.fn();
    const renderer = render(
      <ActionButton variant="primary" active={false} iconName="play" style={{}} fn={fn} />,
    );
    const root = renderer.root;
    expect(root).toBeUndefined();
  });

  it("calls the function when pressed", () => {
    const fn = jest.fn();
    const { getByTestId } = render(
      <ActionButton variant="primary" active={true} iconName="play" style={{}} fn={fn} />,
    );
    const button = getByTestId("action-button");
    fireEvent.press(button);
    expect(fn).toHaveBeenCalledTimes(1);
  });
});
