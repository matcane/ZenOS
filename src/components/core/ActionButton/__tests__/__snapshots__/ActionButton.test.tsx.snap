// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton component renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#2196f3",
      "borderRadius": 999,
      "elevation": 5,
      "height": 64,
      "justifyContent": "center",
      "opacity": 1,
      "position": "absolute",
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.8,
      "shadowRadius": 2,
      "width": 64,
    }
  }
  testID="action-button"
>
  <Text />
</View>
`;
